app/views/listings/index.html.haml

```diff
@@ -17,7 +17,8 @@
 
                 %address.card-text.mb-0= truncate "#{listing.address}, #{listing.city}, #{listing.state}", length: 35
                 .card-text
-                  = link_to "Edit", edit_listing_path(listing), class: "edit-link mt-0", style: "color: #f1ce35; text-decoration: underline"
+                  - if current_user && listing.editable_by?(current_user)
+                    = link_to "Edit", edit_listing_path(listing), class: "edit-link mt-0", style: "color: #f1ce35; text-decoration: underline"
 
```

Comment:
Not sure why, but when this link disappears the css/html formatting gets really terrible